sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/FailureSafeParser.scala

```diff
@@ -65,6 +65,9 @@ class FailureSafeParser[IN](
         case DropMalformedMode =>
           Iterator.empty
         case FailFastMode =>
+          if (e.getCause.getMessage.startsWith("[MALFORMED_RECORD_IN_PARSING")) {
```

Comment:
In this way, determine whether the thrown exception is a subClass of `MALFORMED_RECORD_IN_PARSING`

Reply:
`BadRecordException` should be `SparkThrowable`. Just fill in needed fields and re-throw it.